### PR TITLE
add check for fuel

### PR DESCRIPTION
**Pull request by tikitavi**

### CODE REVIEW

File: juju/fuel/functions [hunk 1]
```diff
@@ -197,19 +203,69 @@ function configure_cluster() {
...
+
 #TODO: remove duplication of code: re-use the check functions from common code when it appears
 function check_storage_pool {
     local mch=${machines[$1]}
     local key_to_check=$2
     local expected_value=$3
 
-    local sp=`config-get storage-pools`
-    local pd=`config-get protection-domain`
+    local sp=`get_config " storage-pools"`
```
Comment (Andrey-mp) [inline juju/fuel/functions]: why quotes and space is needed here?

Comment (tikitavi) [inline juju/fuel/functions]: to prevent confusion with cached-storage-pools

File: juju/fuel/deploy-fuel.sh [hunk 2]
```diff
@@ -5,24 +5,55 @@ my_dir="$(dirname $my_file)"
...
 source $my_dir/functions
 
-
 # provision machines
 provision_machines 0 1 2 3
 
 # prepare fuel master
 prepare_fuel_master 0
 
+disk1="/dev/xvdb"
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel.sh]: note + todo

File: juju/fuel/functions [hunk 3]
```diff
@@ -12,204 +12,303 @@ cd fuel-charms
...
-            juju ssh $i "sudo sed -i 's/\(juju-\)\(.*\)\(machine.*\)/\1\3/g' /etc/hostname && sudo hostname -F /etc/hostname" 2>/dev/null
-        done
-    fi
+  # if not EC2:
+  #       - prepare device for SDS (truncated file)
+  #       - patch host name - in GCE hostname is too long, e.g. juju-023fae94-ef89-4b6d-8681-d9e462c7fae3-machine-2
+  if [[ "${cloud}" == "gce" ]]; then
+    for i in ${new_mx[@]}; do
+      paths=$(echo ${device_paths} | sed 's/,/ /g')
+      juju ssh $i "for p in ${paths}; do sudo rm -f \$p && sudo truncate --size 100G \$p; done"
```
Comment (Andrey-mp) [inline juju/fuel/functions]: revert dev/null

File: juju/fuel/functions [hunk 4]
```diff
@@ -12,204 +12,303 @@ cd fuel-charms
...
+  username='admin'
+  password=`get_config fuel-master password`
+
+  login_cmd="scli --login --username $username --password $password --approve_certificate"
+  sds_names=(`juju ssh $mch "$login_cmd && scli --query_all_sds"| grep 'SDS ID:' | awk '{print$5}'`)
+
+  for sds_name in ${sds_names[@]} ; do
+    output_sds=`juju ssh $mch "$login_cmd && scli --query_sds --sds_name $sds_name"`
+    if ! echo "$output_sds" | grep "$path_type" | grep -q 'total 1 devices'  ; then
+      echo "ERROR: ($my_name:$LINENO) Unexpected number of devices."
```
Comment (Andrey-mp) [inline juju/fuel/functions]: remove $my_name

File: juju/functions [hunk 5]
```diff
@@ -137,3 +137,9 @@ function create_machine() {
     juju machine add --constraints "$additional_constraints instance-type=$instance_type root-disk=40G" 2>&1 | awk '{print $3}'
   fi
 }
+
+function get_config() {
```
Comment (Andrey-mp) [inline juju/functions]: call this from get_cluster_mode (or something)

**Pull request merged**